ffi/custom_input_h.lua

```diff
@@ -0,0 +1,19 @@
+local ffi = require("ffi")
+
+-- Custom values we use for ev.type, ev.code & ev.value in Input and Device HAL implementations
+-- NOTE: EV_SDL == 'S'
+--       MSC_GYRO == 'G'
+
+-- In particular, orientation constants, usable by Device implementations for platform-specific gyro translations.
+-- (matches framebuffer constants, which matches Linux <input/fb.h> FB_ROTATE_* constants).
+-- (i.e., this is how the device is *physically* oriented).
+ffi.cdef[[
+static const int EV_SDL = 83;
+
+static const int MSC_GYRO = 71;
+
+static const int DEVICE_ORIENTATION_UPRIGHT = 0;
+static const int DEVICE_ORIENTATION_CLOCKWISE = 1;
+static const int DEVICE_ORIENTATION_UPSIDE_DOWN = 2;
+static const int DEVICE_ORIENTATION_COUNTER_CLOCKWISE = 3;
```

Comment:
Question: is there a reason we name these differently from the ones that are in screen? Naming them differently requires extra mental mapping IMO.

Reply:
Mainly because I'm not entirely happy with the framebuffer ones ;p.

But,byeah, probably too late for that, will tweak this evening.

Reply:
I went with the nuclear option and actually renamed the Screen ones, mwahahahaha :}.

Reply:
Is there some kind of internal kernel name btw or just the 0, 1, 2, 3?

Reply:
Yup, in `<linux/fb.h>`:

https://github.com/torvalds/linux/blob/b6bb9676f2165d518b35ba3bea5f1fcfc0d969bf/include/uapi/linux/fb.h#L231-L237

Reply:
Which is why you see me peppering UR, CW, UD, CCW comments everywhere ;).

Reply:
> I went with the nuclear option and actually renamed the Screen ones, mwahahahaha :}.

Good God... The man has gone insane... :P